lib/client/client.go

```diff
@@ -1757,6 +1757,15 @@ func (c *NodeClient) ExecuteSCP(ctx context.Context, cmd scp.Command) error {
 	}
 	defer s.Close()
 
+	// File transfers in a moderated session require these two variablesto check for
+	// approval on the ssh server. If they exist in the context, set them in our env vars
+	if moderatedSessionID, ok := ctx.Value(scp.ModeratedSession).(string); ok {
+		s.Setenv(ctx, scp.ModeratedSession, moderatedSessionID)
+	}
+	if fileTransferRequestID, ok := ctx.Value(scp.FileTransferRequest).(string); ok {
+		s.Setenv(ctx, scp.FileTransferRequest, fileTransferRequestID)
+	}
+
```

Comment:
nit: Should we add an error here if only one of these two vars is supplied?

Reply:
Good call. I won't do it here tho, I think in the api handler is better to catch it early on. Thanks!